ecospheres/migrations/20250106_1_themes_as_tags.py

```diff
@@ -0,0 +1,139 @@
+import json
+
+from pathlib import Path
+
+import yaml
+
+from minicli import cli, run
+from slugify import slugify
+from typing import Literal
+
+from ecospheres.api import DatagouvfrAPI
+
+
+FilterTypes = Literal["theme", "subtheme"]
+
+
+def compute_tag(value: str, prefix: FilterTypes) -> str:
+    return slugify(f"ecospheres-{prefix}-{value.lower()}")
+
+
+def find_slug(
+    filter: FilterTypes,
+    value: str,
+    themes_file: Path = Path("ecospheres/migrations/data/themes_new.yaml")
+) -> str | None:
+    with themes_file.open() as f:
+        filters = yaml.safe_load(f)["filters"]["bouquets"]["items"]
+
+    try:
+        values = next(f for f in filters if f["id"] == filter)["values"]
+        filter_value = next(v for v in values if v["name"] == value)
+        return filter_value["id"]
+    except StopIteration:
+        print(f"No slug found for {filter} / '{value}', skipping.")
+        return None
+
+
+@cli
+def migrate_bouquets(slug: str = "", dry_run: bool = False, move: bool = False, env: str = "demo", clean_tags: bool = False):
+    """
+    Copies or move themes/subthemes from extras to slugified tags.
+    Idempotent when using `move=False`.
+    """
+    api = DatagouvfrAPI(env)
+    bouquets = api.get_bouquets()
+    bouquets = [b for b in bouquets if b["slug"] == slug] if slug else bouquets
+    for bouquet in bouquets:
+        print(f"--> Handling {bouquet['slug']}...")
+        original_theme = bouquet["extras"]["ecospheres"].get("theme")
+        original_subtheme = bouquet["extras"]["ecospheres"].get("subtheme")
+        if not original_theme or not original_subtheme:
```

Comment:
Could we have (broken?) cases with only a theme?

Reply:
We shouldn't. But at least we have a warning.